src/internal/connector/graph_connector.go

```diff
@@ -265,9 +266,14 @@ func (gc *GraphConnector) RestoreDataCollections(
 		deets  = &details.Builder{}
 	)
 
+	creds, err := acct.M365Config()
+	if err != nil {
+		return nil, errors.Wrap(err, "malformed azure credentials")
+	}
+
 	switch selector.Service {
 	case selectors.ServiceExchange:
-		status, err = exchange.RestoreExchangeDataCollections(ctx, gc.Service, dest, dcs, deets)
+		status, err = exchange.RestoreExchangeDataCollections(ctx, creds, gc.Service, dest, dcs, deets)
```

Comment:
There should be consistency across the calls:
`ctx, service, dest, dcs, deets` with the same return.

Reply:
See other comment about design transition.